### ISSUE TITLE
Minor issue: NullReferenceException in ConfigurationStore when no ConnectionString is specified

### ISSUE THREAD

**Issue by Kricke242** [Finbuckle/Finbuckle.MultiTenant#376]
I have a service without connection strings. And when I leave the connection string out of the appsettings.config I get a NullReferenceException

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Finbuckle.MultiTenant.Stores.ConfigurationStore`1.UpdateTenantMap()
   at Finbuckle.MultiTenant.Stores.ConfigurationStore`1..ctor(IConfiguration configuration, String sectionName)
   at Finbuckle.MultiTenant.Stores.ConfigurationStore`1..ctor(IConfiguration configuration)
```

Not a big issue, because it works when adding

```
  "Finbuckle:MultiTenant:Stores:ConfigurationStore": {
    "Defaults": {
      "ConnectionString": null
    },
    "Tenants": ...
```

However, it took me a while to figure out what was wrong, since I do not have any database connections in this service, and then I don't think one should have to specify a null one... :-)
Besides, I don't think anything should ever throw a NullReferenceException but rather an ArgumentNullException or similar if the argument is in fact needed. But in this case, I think one should be able to leave the ConnectionString and Default section out altogether...

Thank you, great work!

**Comment by AndrewTriesToCode** [comment]
@Kricke242 Thanks for pointing this out. I agree it should behave a little better. I'm going to mark as a bug so that I remember to address this.

**Comment by natelaff** [comment]
Ah, I just hit this one too :) #325?

**Comment by AndrewTriesToCode** [comment]
@natelaff thanks for chiming in. I'm working on the next release and I'm trying to get CI/CD stuff setup -- a lot of upfront work that I'm hoping will make fixes like this a lot quicker

**Comment by AndrewTriesToCode** [comment]
This is related to #426 and a fix was just merged in #449 so that the defaults section is not required.
